app/controllers/orders_controller.rb

```diff
@@ -0,0 +1,49 @@
+class OrdersController < ApplicationController
+  before_action :logged_in_user
+  before_action :init_cart
+  before_action :load_products
+  before_action :check_product_quantity
+
+  def new
+    @order = current_user.orders.build
+  end
+
+  def create
+    @order = current_user.orders.new order_params
+    if @order.valid?
+      create_transaction
+    else
+      render :new
+    end
+  end
+
+  private
+
+  def order_params
+    params.require(:order).permit(Order::ORDER_ATTRS)
+  end
+
+  def build_order_detail
+    @products.each do |item|
+      product_id = item.id.to_i
+      quantity = @carts[product_id.to_s]
+      price = item.price
+      @order.order_details.create!(book_id: product_id, quantity: quantity,
+                                   price: price,
+                                   total_money: quantity * price)
+    end
+  end
+
+  def create_transaction
+    ActiveRecord::Base.transaction do
+      @order.save
```

Comment:
dùng `save!` để nó raise exception
```suggestion
      @order.save!
```